File: InMotionGIT.Common - Copy/Helpers/AplicationSetting.vb
Imports System.Configuration

Namespace Helpers

    <Obsolete("Not used any more, se migro a las extencions AppSettingsExtensions", True)>
    Public NotInheritable Class ApplicationSetting

        Public Shared Function IsTrue(appSettingName As String) As Boolean
            Dim result As Boolean = False
            If ConfigurationManager.AppSettings(appSettingName).IsNotEmpty AndAlso
               ConfigurationManager.AppSettings(appSettingName).ToLower.Equals("true") Then
                result = True
            End If

            Return result
        End Function

        Public Shared Function IsFalse(appSettingName As String) As Boolean
            Dim result As Boolean = True
            If ConfigurationManager.AppSettings(appSettingName).IsNotEmpty AndAlso
               ConfigurationManager.AppSettings(appSettingName).ToLower.Equals("false") Then
                result = True
            End If

            Return result
        End Function

        Public Shared Function StringValue(appSettingName As String) As String
            Return StringValue(appSettingName, String.Empty)
        End Function

        Public Shared Function StringValue(appSettingName As String, defaultValue As String) As String
            Dim result As String = String.Empty
            If ConfigurationManager.AppSettings(appSettingName).IsNotEmpty Then
                result = ConfigurationManager.AppSettings(appSettingName)
            End If
            If result.IsEmpty Then
                result = defaultValue
            End If

            Return result
        End Function

    End Class

End Namespace

File: Common/InMotionGIT.Common/Helpers/AplicationSetting.cs
using System;
using System.Configuration;
using InMotionGIT.Common.Extensions;

namespace InMotionGIT.Common.Helpers
{

    [Obsolete("Not used any more, se migro a las extencions AppSettingsExtensions", true)]
    public sealed class ApplicationSetting
    {

        public static bool IsTrue(string appSettingName)
        {
            bool result = false;
            if (ConfigurationManager.AppSettings[appSettingName].IsNotEmpty() && ConfigurationManager.AppSettings[appSettingName].ToLower().Equals("true"))
            {
                result = true;
            }

            return result;
        }

        public static bool IsFalse(string appSettingName)
        {
            bool result = true;
            if (ConfigurationManager.AppSettings[appSettingName].IsNotEmpty() && ConfigurationManager.AppSettings[appSettingName].ToLower().Equals("false"))
            {
                result = true;
            }

            return result;
        }

        public static string StringValue(string appSettingName)
        {
            return StringValue(appSettingName, string.Empty);
        }

        public static string StringValue(string appSettingName, string defaultValue)
        {
            string result = string.Empty;
            if (ConfigurationManager.AppSettings[appSettingName].IsNotEmpty())
            {
                result = ConfigurationManager.AppSettings[appSettingName];
            }
            if (result.IsEmpty())
            {
                result = defaultValue;
            }

            return result;
        }

    }

}